### DISCUSSION TITLE
MCA related question

### DISCUSSION THREAD

**Discussion by NouraAlkhaja** [pysal/momepy#506] [Q&A]
Hey! Thank you so much for this super useful tool, I am yet to explore its full potential. I have few questions though, that I would be grateful if I could have an answer for:
1- Its not quite clear yet as to how the weight is used in the centrality measures. In the manual you mention that:
"_then radius=400 and distance=length of each segment saved as a parameter of each edge. By default, momepy saves length as mm_len.The weight parameter is used for centrality calculation. Again, we can use metric weight (using the same attribute as above) or no weight (weight=None) at all. Or any other attribute we wish._"

- the length of each segment is saved as parameter for what purpose? it will be assigned as an attribute to each edge?
- I know that the calculation for Closeness for example would calculate the distance to all nearby nodes within the radius and then average them for each node. This average  will be the result assigned to that node. Right? So the language used in the manual is not clear as to what the "weight" or the " edge length saved as parameter" refers to? 
- Is the purpose of weight is to weigh the resulting average as per the summed lengths of the edges connecting each node with the nearby nodes? what if no weight is used as the manual mentions? 
- same questions go to the Dual graph. I hope the manual can be revised to explain better what are the formulas used for each measure? what is the edge parameter and how its used? what is the weight and how its used?
- more detailed documentation to this function is also required please: "_dual_gdf.plot(ax=ax, column='angcloseness400', cmap='Spectral_r', scheme='quantiles', k=15, alpha=0.6)_"

In the UNA toolbox for example, used in Rhino 3D, the weight is assigned to origins or destinations to impact the result. Bigger destinations would be assigned more trips for example. 

2- Would it be possible to utilize graphs from shapefiles rather than OSM? 

Tons of gratitude for this open source tool!

**Comment by martinfleis** [comment]
- `weight` is used when computing the shortest path for the centrality measurement.
- when `radius` is used, you are computing the local version of centrality. That means that for each node, you are defining a subgraph of nodes within a set radius. This radius can be weighted independently of `weight` by using `distance`. This means that you can measure local centrality based on time by doing `weight="time"` on local subgraphs defined as 400 meters with `radius=400, distance="lenght"`, assuming you have edge attributes called `"time"` and `"length"`.

> the length of each segment is saved as parameter for what purpose? it will be assigned as an attribute to each edge?

because most of graph-based metrics implemented in momepy are primarily based on metric distance. So it is handy not to think about creating one yourself prior to conversion from GeoDataFrame to Graph. And yes, it is assigned as edge attribute.

> This average will be the result assigned to that node. Right? So the language used in the manual is not clear as to what the "weight" or the " edge length saved as parameter" refers to?

Yes. Do you have a suggestion how to reformulate it?

> Is the purpose of weight is to weigh the resulting average as per the summed lengths of the edges connecting each node with the nearby nodes? what if no weight is used as the manual mentions?

No, the purpose of weight is to give proper geographical weight to each edge within the shortest distance calculation. If no weight is given, distance is only topological.

>  I hope the manual can be revised to explain better what are the formulas used for each measure? what is the edge parameter and how its used? what is the weight and how its used?

Formulas are the same for primal and dual. I recommend reading relevant literature or proposing how we should change our documentation. We're offering a tool and expect users to learn the conceptual backbone in literature.

> more detailed documentation to this function is also required please: "`dual_gdf.plot(ax=ax, column='angcloseness400', cmap='Spectral_r', scheme='quantiles', k=15, alpha=0.6)"`

This is a standard GeoPandas plotting code. Please refer to GeoPandas documentation.

> In the UNA toolbox for example, used in Rhino 3D, the weight is assigned to origins or destinations to impact the result. Bigger destinations would be assigned more trips for example.

This is not supported by momepy. 

> Would it be possible to utilize graphs from shapefiles rather than OSM?

See http://docs.momepy.org/en/stable/user_guide/graph/convert.html
